camelot/admin/action/list_action.py

```diff
@@ -371,6 +371,13 @@ def model_run( self, model_context ):
         yield action_steps.UpdateObjects(updated_objects)
         yield action_steps.FlushSession(model_context.session)
 
+    def get_state(self, model_context):
+        state = super().get_state(model_context)
+        if isinstance(model_context, ListActionModelContext):
```

Comment:
is dit niet altijd een ListActionModelContext ?

Reply:
Er was ook nog een call met een FieldActionModelContext. Deze gebeurt in controls/delegates/customdelegate.py op lijn 156 in functie get_standard_item. Deze zet de ActionStatesRole data die dan gebruikt w in setEditorData.

Voor de One2ManyEditor is deze eigenlijk overbodig nu. Ik heb een extra parameter with_action_states=True toegevoegd aan get_standard_item om al dan niet de ActionStatesRole data aan te maken. De One2ManyDelegate roept deze functie dan aan met with_action_states=False.

Reply:
Negeer dit aub. Ik zal in de volgende PR meer uitloeg geven.